src/components/pages/Prices/Prices.tsx

```diff
@@ -0,0 +1,159 @@
+import TableView from '@src/components/data-display/tables/TableView';
+import { useDemandFactor } from '@src/hooks/useDemandFactor';
+import {
+  camelToReadable,
+  formatARIOWithCommas,
+  lowerCaseDomain,
+} from '@src/utils';
+import {
+  ARNS_NAME_REGEX_PARTIAL,
+  MAX_ARNS_NAME_LENGTH,
+} from '@src/utils/constants';
+import eventEmitter from '@src/utils/events';
+import { createColumnHelper } from '@tanstack/react-table';
+import { useEffect, useState } from 'react';
+
+type TableData = {
+  characters: number;
+  baseFee: number;
+  adjustedRegistrationFee: number;
+  renewalFee: number;
+  oneYearLeaseFee: number;
+  permabuyFee: number;
+  undernameIncreaseFee: number;
+  permabuyUndernameIncreaseFee: number;
+} & Record<string, any>;
+
+const columnHelper = createColumnHelper<TableData>();
+
+const minFees = new Array(MAX_ARNS_NAME_LENGTH - 12)
+  .fill(true)
+  .reduce((acc, _, index) => {
+    acc[index + 13] = 200;
+    return acc;
+  }, {});
+
+const BASE_NAME_FEES: Record<number, number> = {
+  1: 1000000,
+  2: 200000,
+  3: 20000,
+  4: 10000,
+  5: 2500,
+  6: 1500,
+  7: 800,
+  8: 500,
+  9: 400,
+  10: 350,
+  11: 300,
+  12: 250,
+  ...minFees,
+};
+
+export default function Prices() {
+  const { data: demandFactor, isLoading, error } = useDemandFactor();
+  const [domain, setDomain] = useState('');
+
+  const [tableData, setTableData] = useState<TableData[]>([]);
+
+  useEffect(() => {
+    if (!demandFactor) return;
+    const newData = Object.entries(BASE_NAME_FEES).map(
+      ([characters, baseFee]) => {
+        const arf = baseFee * demandFactor;
+        const af = arf * 0.2;
+        const data: TableData = {
+          characters: Number(characters),
+          baseFee,
+          adjustedRegistrationFee: arf,
+          renewalFee: af,
+          oneYearLeaseFee: arf + af * 1,
+          permabuyFee: arf + af * 20,
+          undernameIncreaseFee: baseFee * demandFactor * 0.001,
+          permabuyUndernameIncreaseFee: baseFee * demandFactor * 0.005,
+        };
+        return data;
+      },
+    );
+    setTableData(newData);
+  }, [demandFactor, isLoading, error]);
+
+  const columns = [
+    'characters',
+    'baseFee',
+    'adjustedRegistrationFee',
+    'renewalFee',
+    'oneYearLeaseFee',
+    'permabuyFee',
+    'undernameIncreaseFee',
+    'permabuyUndernameIncreaseFee',
+  ].map((key) => {
+    return columnHelper.accessor(key, {
+      id: key,
+      header: camelToReadable(key),
+      cell: ({ row }) => {
+        const numVal = Number(row.getValue(key));
+        if (key === 'characters') return numVal.toString();
+        return formatARIOWithCommas(Number(numVal.toFixed(3)));
+      },
+    });
+  });
+
+  return (
+    <div className="page">
+      <div className="flex w-full py-2 justify-between  items-center gap-5">
```

Comment:
wouldn't mind seeing a little more space betwen this header row and the table (like an mb-2)

Reply:
Validated all styles with jon - this isn't a user-facing UI so we can come back for any more style nits later.